extensions/functions_arithmetic.yaml

```diff
@@ -1563,6 +1563,43 @@ aggregate_functions:
             values: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
         nullability: DECLARED_OUTPUT
         return: fp64?
+  - name: "first"
+    description: >-
+      First value in a set of rows.
+
+      The `null_handling` option determines whether or not null values will be recognized by the function.
+      If `null_handling` is set to `IGNORE_NULLS`, the function will look for the first non-null value.
+      If set to `ACCEPT_NULLS`, the function will return the first value it sees, even if null.
+
+      If all values are null or there are no values, returns null.
+    impls:
+      - args:
+          - name: x
+            value: any
+        options:
+          null_handling:
```

Comment:
I copied this from [concat](https://github.com/substrait-io/substrait/blob/fe7eba4454d404705cd124151856260c4a2b13bf/extensions/functions_string.yaml#L20), does it make sense here or is better to add e.g. a boolean arg?

Reply:
I think options make sense for this. The names `ACCEPT_NULLS` and `IGNORE_NULLS` sound a little weird to me, but I can't think of better ones currently.

Reply:
I added that original option to concat, but I agree it does sound a little weird.  How about changing the option name to `ignore_nulls` and have the options be `["True", "False"]`.  I think True/False may need to be quoted.  If i recall correctly I didn't do this originally because no other options were quoted, but that's changed since then.

Reply:
renamed in https://github.com/substrait-io/substrait/pull/652/commits/b18cecd00aee3aeab891f4e095c9ee25942be255 - and quoted after, as that was needed to keep them as strings. Though it hurts my soul a bit to have a string "TRUE". Maybe should make them YES/NO instead to be less confusing 😅